services/device-registry-base/src/main/java/org/eclipse/hono/service/management/device/Device.java

```diff
@@ -280,4 +283,23 @@ public String getMapper() {
         return mapper;
     }
 
+    /**
+     * Sets the registry internal status information of this device.
+     *
+     * @param status The status information to be set or {@code null} if there is none.
+     * @return A reference to this for fluent use.
+     */
+    public Device setStatus(final Status status) {
+        this.status = status;
+        return this;
+    }
+
+    /**
+     * Gets the registry internal status information of this device.
+     *
+     * @return The registry internal status information or {@code null} if there is none.
+     */
+    public Status getStatus() {
```

Comment:
> I don't know if I have covered everything but it is very hard to review this PR as so many changes are involved all over the place. IMHO it would be better to split this PR up into smaller parts, maybe based on the three different registry implementations? We could then discuss the overall idea and concept in a smaller scope first, before rolling it out to all implementations.
> One thing that struck my mind in particular is that of DB migration, i.e. can a user update to this new code and still use his/her existing DB content?

The changes should be backward compatible for the MongoDb based registry (see https://github.com/eclipse/hono/pull/2189#issuecomment-703703151). *But: * I would like to run a final test after having rebased on master again, just to be sure.

I _think_ this should also be the case for the JDBC Registry, since it also only stores a JSON string but I would really appreciate feedback from @ctron on that 😃